### DISCUSSION TITLE
Make colsep part of the column width

### DISCUSSION THREAD

**Discussion by al3atross** [lvjr/tabularray#297] [Ideas]
### Problem
I'd like to have a table with column ratio of 3:1.

`X` columns seem to be the right approach as they occupy the remaining width in proportion to their coefficients.

Unfortunately, the `colsep` dimension is not part of the column width.
That's why it is subtracted from the remaining width before the column widths are determined.

This results in different positions of the **Vline** depending on the `colsep` as the following example illustrates.

```latex
\documentclass{article}
\usepackage{tabularray}

\begin{document}

\begin{tblr}{colspec={|X[3]|X[1]|}, hlines, colsep=0pt}
	A & B \\
\end{tblr}

\vspace{3mm}
	
\begin{tblr}{colspec={|X[3]|X[1]|}}
	A & B \\
\end{tblr}

\vspace{3mm}

\begin{tblr}{colspec={|X[3]|X[1]|}, hlines, colsep=24pt}
	A & B \\
\end{tblr}

\end{document}
```

![image](https://user-images.githubusercontent.com/1773071/185950606-e4a02e6d-3d71-4d17-b163-2ff01e7b3d7d.png)

While the first table (with `colsep=0pt`) has a Vline-to-Vline ratio of 3:1, the last table (with `colsep=24pt`) has a Vline-to-Vline ratio of about 2.3:1.

### Proposal

My proposal is to include the 'colsep' into the column width such that the Vline-to-Vline ratio is maintained.
(There might be other dimensions like `rulewidth` that have to be taken into account to achieve the exact ratio.)

**Comment by lvjr** [comment]
As I have commented below another [post](https://github.com/lvjr/tabularray/discussions/292) of yours, you should not use `X` columns if you need fixed-width ciolumns:

```latex
\documentclass{article}
\usepackage{tabularray}
\usepackage{calc}
\begin{document}

\newlength\lengtha \setlength\lengtha{0.75\linewidth-0.4pt}
\newlength\lengthb \setlength\lengthb{0.25\linewidth-0.4pt}

\begin{tblr}{colspec={|Q[\lengtha]|Q[\lengthb]|}, hlines, colsep=0pt}
	A & B \\
\end{tblr}

\vspace{3mm}

\newlength\lengthc \setlength\lengthc{0.75\linewidth-0.4pt-12pt}
\newlength\lengthd \setlength\lengthd{0.25\linewidth-0.4pt-12pt}

\begin{tblr}{colspec={|Q[\lengthc]|Q[\lengthd]|}, hlines}
	A & B \\
\end{tblr}

\vspace{3mm}

\newlength\lengthe \setlength\lengthe{0.75\linewidth-0.4pt-48pt}
\newlength\lengthf \setlength\lengthf{0.25\linewidth-0.4pt-48pt}

\begin{tblr}{colspec={|Q[\lengthe]|Q[\lengthf]|}, hlines, colsep=24pt}
	A & B \\
\end{tblr}

\end{document}
```

![image](https://user-images.githubusercontent.com/4239949/186060575-5d8f9c49-0af7-4e7f-b055-4aa6f9904d45.png)
